sdk/keyvault/keyvault/src/az_keyvault_client.c

```diff
@@ -137,13 +135,13 @@ AZ_NODISCARD az_result az_keyvault_keys_key_create(
 
   // add version to request
   AZ_RETURN_IF_FAILED(az_http_request_builder_set_query_parameter(
-      &hrb, AZ_STR("api-version"), client->retry_options.service_version));
+      &hrb, AZ_STR("api-version"), AZ_KEYVAULT_API_VERSION));
```

Comment:
Another option here is to create an ApiVersion policy and configure it with the AZ_KEYVAULT_API_VERSION option and then ALL operations will just get this header added automatically so there is no need to repeat this code for every operation. You could use the same policy for storage too. In this case, I'd make the query parameter name be an option along with the version string.

Reply:
That's a good idea.  I'll make that change in a subsequent PR.